awx/api/views/__init__.py

```diff
@@ -4350,3 +4350,48 @@ def post(self, request, *args, **kwargs):
             return Response({"error": _("This workflow step has already been approved or denied.")}, status=status.HTTP_400_BAD_REQUEST)
         obj.deny(request)
         return Response(status=status.HTTP_204_NO_CONTENT)
+
+
+class ResourceStateView(GenericAPIView):
```

Comment:
Can you remove the workspace logic from this? The http backend does not currently support workspaces, and I do not want to add workspace support without further discussion about what it should look like and whether or not we even want to do this.